### PR TITLE
Istio resources reconciler

### PR DESCRIPTION
**Pull request by triffer**
**Description**

Changes proposed in this pull request:

- Implement PoC to reconcile Istio resources and restart proxy sidecar and ingress gateway depending on criteria defined by each resource.

**Related issue(s)**

- https://github.com/kyma-project/istio/issues/268

### CODE REVIEW

File: internal/reconciliations/ingress_gateway/reconciliation.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	err := k8sClient.List(ctx, &list, &client.ListOptions{Namespace: namespace, LabelSelector: ls})
+	if err != nil {
+		return nil, err
+	}
+
+	return &list, err
+
+}
+
+// TODO this code is duplicated in istio installation package, we should reuse it
```
Comment (triffer) [inline internal/reconciliations/ingress_gateway/reconciliation.go:84]: We should solve TODOs added in the PR, before merging it.

File: internal/reconciliations/ingress_gateway/reconciliation.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+			}
+		}
+
+		if mustRestart {
+			break
+		}
+	}
+
+	if mustRestart {
+		// TODO: just re-used the existing logic, we can also think about restarting the pod directly
```
Comment (triffer) [inline internal/reconciliations/ingress_gateway/reconciliation.go:58]: We should solve or remove TODOs added in the PR, before merging it.

File: pkg/lib/sidecars/pods/get.go [hunk 3]
```diff
@@ -51,7 +52,24 @@ func GetPodsToRestart(ctx context.Context, c client.Client, expectedImage Sideca
...
+
+		for i, pod := range podList.Items {
+			if evaluator.RequiresProxyRestart(pod) {
+				outputPodsList.Items = append(outputPodsList.Items, *pod.DeepCopy())
+				remaining.Items = append(remaining.Items[:i], remaining.Items[i+1:]...)
+			}
+		}
+	}
+
+	// TODO: Migrate needsRestart to predicate
```
Comment (triffer) [inline pkg/lib/sidecars/pods/get.go:71]: We should solve or remove TODOs added in the PR, before merging it.

**Review by werdes72 — APPROVED** [review]

**Pull request merged**